File: prime_generator.java
import java.util.Scanner;
public class main {
    static boolean primeCheck(int a) {
        int w = 2;
        int i = 2;
        while (i < a) {
            if (a != w) {
                double aOverW = (double) a / w;
                int intAOverW = a / w;
                if (Math.abs(aOverW-intAOverW) < 0.0000001) {
                    return false;
                }
            }
            w ++;
            i ++;
        }
        return true;
    }
    public static void main(String[] args) {
        System.out.println("This Java program outputs primes from 2 to a maximum");
        System.out.println("Enter a maximum");
        Scanner myObj = new Scanner(System.in);
        int maxVar = myObj.nextInt();
        int limitVar = 2;
        while (limitVar<=maxVar) {
            boolean z = primeCheck(limitVar);
            if (z) {
                System.out.println(limitVar);
            }
            limitVar += 1;
        }
    }
}

File: prime_generator.kt
package my.demo
import kotlin.math.abs
fun convertDouble(a: Int): Double {
    return a.toDouble()
}
fun convertDouble2Int(a: Double): Int {
    return a.toInt()
}
fun convertString2Int(a: String): Int {
    return a.toInt()
}
fun primeCheck(a: Int): Boolean {
    var w = 2.0
    var i = 1
    while (i < a) {
        if (a.toDouble() != w) {
            val aOverW: Double = a / w
            val intAOverW = convertDouble2Int(aOverW)
            val doubleIntAOverW = convertDouble(intAOverW)
            if (abs(aOverW-intAOverW) < 0.00000001) {
                return false
            }
        }
        w += 1
        i += 1
    }
    return true
}
fun main() {
        println("This Kotlin program outputs primes from 2 to the maximum")
        println("Enter a maximum")
        val mVar: String = readln()
        val maxVar = convertString2Int(mVar)
        var limitVar = 2
        while (limitVar<=maxVar) {
            val z = primeCheck(limitVar)
            if (z) {
                println(limitVar)
            }
            limitVar += 1
        }
    }
